jupyter-book/introduction/raw_data_processing.md

```diff
@@ -1,52 +1,397 @@
+
+```bash
+CONDA_SUBDIR=osx-64 conda create -n af_xmpl -y -c bioconda python=3.9 salmon alevin-fry pyroe   # create a new environment
+conda activate af_xmpl
+conda env config vars set CONDA_SUBDIR=osx-64  # subsequent commands use intel packages
+```
+
+Next, we create a working directory `af_xmpl_run`, and download and uncompress the example dataset from a remote host.
+
+```bash
+# create a working dir 
+mkdir af_xmpl_run && cd af_xmpl_run
+
+# download the example dataset
+wget -qO- https://umd.box.com/shared/static/lx2xownlrhz3us8496tyu9c4dgade814.gz | tar xzf - --strip-components=1 -C .
+
+# download CB permit list 
+wget https://raw.githubusercontent.com/10XGenomics/cellranger/master/lib/python/cellranger/barcodes/3M-february-2018.txt.gz && gunzip 3M-february-2018.txt.gz
+
+```
+
+(raw-proc:example-map)=
+### Mapping the data
+
+Now, we process the data to obtain the mapping information. This includes three steps:
+1. Build the _splici_ reference (<u>splic</u>ed transcripts + <u>i</u>ntrons), using the genome and gene annotation file.
+2. Index the _splici_ reference.
+3. Map the reads against the _splici_ index.
+
+```bash
+# making splici reference
+# pyroe make-splici genome_file gtf_file read_length out_dir
+pyroe make-splici \
+toy_human_ref/fasta/genome.fa \
+toy_human_ref/genes/genes.gtf \
+90 \
+splici_ref
+
+# Indexing the reference
+# salmon index -t extend_txome.fa -i idx_out_dir -p num_threads
+salmon index \
+-t $(ls splici_ref/*.fa) \
+-i splici_idx \
+-p 8 
+
+# collecting reads files
+fastq_dir="toy_read_fastq"
+reads1_pat="_R1_"
+reads2_pat="_R2_"
+reads1="$(find -L $fastq_dir -name "*$reads1_pat*" -type f | xargs | sort | awk '{print $0}')"
+reads2="$(find -L $fastq_dir -name "*$reads2_pat*" -type f | xargs | sort | awk '{print $0}')"
+
+# Mapping
+# salmon alevin -i idx_out_dir -l library_type -1 reads1_files -2 reads2_files -p num_threads -o map_out_dir
+salmon alevin \
+-i splici_idx \
+-l ISR \
+-1 ${reads1} \
+-2 ${reads2} \
+-p 8 \
+-o alevin_map \
+--chromiumV3 \
+--sketch
+```
+
+This will produce an output folder called `alevin_map` that contains all the information we need to process the mapped reads using `alevin-fry`. 
+
+(raw-proc:example-quant)=
+### Processing the mapped reads
+
+Next, we run the cell barcode correction, UMI resolution, and quantification steps described previously using `alevin-fry`. This procedure involves three `alevin-fry` commands:
+
+1. The `generate-permit-list` command is used for cell barcode correction.
+2. The `collate` command filters out invalid mapping records, corrects cell barcodes, and collates mapping records originating from the same cell.
+3. The `quant` command performs UMI resolution and quantification.
+
+```bash
+# cell barcode correction
+# alevin-fry generate-permit-list -u CB_permit_list -d expected_orientation -o gpl_out_dir 
+alevin-fry generate-permit-list \
+-u 3M-february-2018.txt \
+-d fw \
```

Comment:
Maybe I missed it, but if the read orientation was not mentioned before it might be a good idea to mention this.

Reply:
Good point. We added a paragraph in the mapping/alignment section for read orientation.